lib/rules/object-curly-newline.js

```diff
@@ -88,6 +88,38 @@ function normalizeOptions(options) {
     return { ObjectExpression: value, ObjectPattern: value };
 }
 
+/**
+ * Gets the closing brace of an object expression with support for flow-style type annotations
+ *
+ * @param {SourceCode} sourceCode - The SourceCode object for the current object node
+ * @param {ASTNode} openBrace - The open brace for the object node
+ *
+ * @returns {ASTNode|null} The closing brace, or null if none is found
+ *
+ */
+function getMatchingClosingBrace(sourceCode, openBrace) {
+
+    // With flow, an object can have the form of `{ a : 3 } : { a : number }`. In this form the
+    // closing brace isn't simply the last brace. In order to find the closing brace we actually
+    // need to traverse through the node and count the braces, to find the matching pair.
```

Comment:
Is there a way to do this that doesn't involve linearly traversing the node and counting braces? I'm imagining something like this:

```js
if (node.typeAnnotation) {
    return sourceCode.getTokenBefore(node.typeAnnotation);
} else {
    return sourceCode.getLastToken(node);
}
```

Reply:
I vaguely remember looking for that, but couldn't find it. Nevertheless, I'll look again and see if I can make this linear.

Reply:
Welp, that exact snippet seems to have done the trick :smile: Thanks for helping me out!